product-scenarios/5-route-messages-between-systems/5.1-Route-based-on-the-content-of-the-messages/5.1.1-Route-messages-based-on-the-given-xpath/5.1.1.3-Using-filter-mediator-on-given-xpath/src/test/java/org/wso2/carbon/esb/scenario/test/RoutingBasedOnXpathWithFilterMTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.scenario.test;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.carbon.esb.scenario.test.common.http.HttpConstants;
+
+/**
+ * This test class tests routing messages into relevant backends based on the XPath by using filter mediator.
+ */
+
+public class RoutingBasedOnXpathWithFilterMTest extends ScenarioTestBase {
+
+    @BeforeClass
+    public void init() throws Exception {
+        super.init();
+    }
+
+    @Test(description = "5.1.1.3.1")
+    public void testRouteBasedOnXpathWithValidCaseName() throws Exception {
+        String header = "5_1";
+        String url = getApiInvocationURLHttp("5_1_API_Routing_messages_based_on_content_of_message_test" +
+                "/valid_xpath_test_with_filterM");
+        String request = "<m:GetStockPrice xmlns:m=\"http://www.example.org/stock\">\n"
```

Comment:
IMO, we can move the message to ScenarioConstants so that it can be used everywhere. WDYT?